tooling/bin/assess-minibroker-charts

```diff
@@ -0,0 +1,158 @@
+# Add new engines here, and at (x).
+
+bin="${TOP}/tooling/mb-chart-assessment"
+wrk="${TOP}/_work"
+stem=kube-chart-index
+rstem=dbe
+master="${wrk}/${rstem}-charts-master-index.yaml"
+
+stable=https://kubernetes-charts.storage.googleapis.com
+index="${stable}/index.yaml"
+
+nodeenv=src/scf-release/src/acceptance-tests-brain/test-resources/node-env
+
+# Setup ................................................................
+
+if [ $INCREMENTAL -eq 0 ] ; then
+    echo $(cyan Cleared) previous run, starting from empty
+    mkdir -p "${wrk}"
+    # Clear leftovers from previous runs
+    rm -rf ${wrk}/${rstem}-*
+else
+    echo $(cyan Incremental) run, keeping data, will skip processed charts
+fi
+
+# 1. Get master index .....................................................
+
+echo $(cyan Retrieving) master index ...
+curl --silent "${index}" > "${master}"
+
+# 2. Extract the full data for all the supported engines ..................
+
+for engine in $engines
+do
+    echo -n $(cyan Extracting) engine $(blue "${engine}") ...
+    # Originally attempted to use a pipeline using `y2j | jq ...`.
+    # y2j failed on the incoming yaml, claiming a number of things as
+    # not serializable to json.
+    ruby "${bin}/get-charts-for-engine.rb" "${engine}" "${master}" \
+	 > "${wrk}/${rstem}-${engine}.yaml"
+
+    # 2.a
+    echo -n "$(left 4)$(eeol), listing ..."
+    ruby "${bin}/get-chart-listing.rb" "${wrk}/${rstem}-${engine}.yaml" \
+	 > "${wrk}/${rstem}-${engine}-charts.csv"
+    # CSV using space as field separator (-> processable by shell, see below)
+    echo -n "$(left 13)$(eeol): $(cyan "$(cat "${wrk}/${rstem}-${engine}-charts.csv" | wc -l)")"
+    echo " $(green OK)"
+done
+
+#.......................................................................
+#exit
+
+# 3. Iterate over engines & charts, capture pass/fail for test ............
+
+if [ $INCREMENTAL -eq 0 ] ; then
+    rm -f "${wrk}/dbe-chart-results.csv"
+fi
+
+echo $(cyan Testing) ... Charts: $(cyan "$(cat  ${wrk}/${rstem}-*-charts.csv  | wc -l)")
+
+skipped=0
+for engine in $engines
+do
+    while read enginev chartv chartlocation
+    do
+	if [ $INCREMENTAL -gt 0 ] ; then
+	    # In incremental mode, skip all processed charts.
+
+	    if grep -q "${engine} ${enginev} ${chartv}" "${wrk}/${rstem}-chart-results.csv" ; then
+		echo -en "\r$(eeol)Skipping ${engine}" "${enginev}" "${chartv}"
+		skipped=$(( $skipped + 1 ))
+		continue
+	    fi
+	fi
+	echo -en "\r$(eeol)"
+
+	mkdir -p "${wrk}/${rstem}-${engine}"
+
+	# 2.b Generate a helm repo index for the specific engine and chart.
+	ruby "${bin}/make-chart-index.rb" "${engine}" "${enginev}" "${chartv}" "${wrk}/${rstem}-${engine}.yaml" \
+	     > "${wrk}/${rstem}-${engine}/${enginev}-${chartv}.yaml" 2>&1
+
+	"${bin}/assess-single-chart" \
+	    "${CF_NAMESPACE}" "${engine}" "${enginev}" "${chartv}" "${chartlocation}" "${CAP}" \
+	    || true
+	# Failures are ignored in favor of checking more charts.
+	# The logs can be looked at later.
```

Comment:
Don't really need the output, the `assess-single-chart` has already reported that.
I.e. this would be 

```
if ! "${bin}/assess-single-chart" … ; then
  : # Failed to assess, ignore
  # maybe a continue
fi
```

Reply:
Left as is for now. Will see how it looks when going to ruby.